python/setup.py

```diff
@@ -189,6 +193,10 @@ def _run_cmake(self):
 
             if self.cmake_generator:
                 cmake_options += ['-G', self.cmake_generator]
+            if self.with_cuda:
+                cmake_options.append('-DPYARROW_BUILD_CUDA=on')
+            else:
+                cmake_options.append('-DPYARROW_BUILD_CUDA=off')
```

Comment:
Removed  `-DPYARROW_BUILD_CUDA=off` as redundant.
Detecting Arrow CUDA is implemented in FindArrowCuda.cmake. Moving the detection algorithm to setup.py is involved and against the current logic in cmake/setup files, imho.

An option is to make `--with-cuda` default, that is, replace it with `--without-cuda`.

Reply:
I guess it would be possible to analyze the output of cmake to detect if libarrow_gpu library is available and then define with_cuda flag. It sounds hackish though.

Reply:
Why not try to build and fail silently, as for Parquet and Orc?

Reply:
setup.py-wise, _cuda is configured and build exactly like _parquet or _orc. So I don't understand your analogue, especially because parquet and orc are disabled by default.

I suggest the following behavior:
Case 1: libarrow_gpu can be detected by cmake, then
- `python setup.py build_ext --with-cuda` will build `_cuda` extension
- `PYARROW_BUILD_CUDA=1 python setup.py build_ext` will build `_cuda` extension
- `python setup.py build_ext` will succeed, no `_cuda` extension is built
- `PYARROW_BUILD_CUDA=0 python setup.py build_ext` will succeed, no `_cuda` extension is built

Case 2: libarrow_gpu is not detected by cmake, then
- `python setup.py build_ext --with-cuda` will fail
- `PYARROW_BUILD_CUDA=1 python setup.py build_ext` will fail
- `python setup.py build_ext` will succeed, no `_cuda` extension is built